vscode/microsoft-kiota/src/extension.ts

```diff
@@ -57,7 +58,59 @@ export async function activate(
   );
   const reporter = new TelemetryReporter(context.extension.packageJSON.telemetryInstrumentationKey);
   await loadTreeView(context);
+  await checkForLockFileAndPrompt(context);
   let codeLensProvider = new CodeLensProvider();
+  async function checkForLockFileAndPrompt(context: vscode.ExtensionContext) {
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+
+    if (workspaceFolders) {
+      let lockFilePath: string | undefined;
```

Comment:
there might be multiple lock files. Is the strategy to prompt people once per lock file here?

Reply:
I think you just need to find at least one lock file(to prove that migration is needed) the pass the workspace folder path. The server should migrate all the lock files to the workspace file.

Reply:
oh yeah! That's fine then